### ISSUE TITLE
Cannot start flanneld after rollback to 1122.3.0 (from 1185.3.0)

### ISSUE THREAD

**Issue by ntquyen** [coreos/bugs#1652]
Got a problem with docker v1.11.2 after upgrading to CoreOS `1185.3.0`, so I rollback to previous one, `1122.3.0`, then flannel cannot start anymore with this bug:

```
rkt[1768]: run: cannot open store: current store db version: 7 (greater than the current rkt expected version: 5)
```

**Comment by crawford** [comment]
It looks like rkt's datastore is not forward compatible. You'll need to clear out the data store before continuing. @lucab would know the details.

Could you file a bug describing the issue you ran into with Docker 1.11.2?

**Comment by ntquyen** [comment]
@crawford Thanks, I deleted the `/var/lib/rkt` and it's working again!
There are several issues when I upgrade to 1185.3.0. The most critical one is that it seem docker is hanging after running for a while (about 12h or more). This only happen on VMs where there are a lot of kubernetes pods being deleted/scheduled. First observation is that all pods is at `Pending` on those nodes, then `docker ps` hangs. This can only be fixed when restart the VM.

Another issue is that kubelet metrics endpoint `http://[host-ip]/10255/metrics` miss all series with `io_kubernetes_` labels.

I'm running kubernetes cluster v1.2.0

**Comment by lucab** [comment]
Thanks for the report @ntquyen. We already had some previous discussion on this on rkt, but we didn't capture it on any github ticket, so I just opened https://github.com/coreos/rkt/issues/3358 for this. As mentioned there, we don't do down-migrations at the moment but we take DB backups before up-migrations (so it may be still available to users, but it has poor UX).

**Comment by brancz** [comment]
Depending on the cadvisor version in use the label names may have changed. Could you check if there are label with `kubernetes_` in your metrics output? Some metrics also need to be set once to show up at all in the metrics output.

**Comment by say5** [comment]
We are having the same problem on 1185.3.0 as it was described by @ntquyen, after some time docker stops working, we are running Kubernetes cluster 1.4.3 on AWS.

**Comment by crawford** [comment]
@ntquyen can you open separate bug reports for each of your distinct issues so we can properly track them? Since the topic of this report has been addressed, I'm going to close it out.

**Comment by crawford** [comment]
> The most critical one is that it seem docker is hanging after running for a while (about 12h or more). This only happen on VMs where there are a lot of kubernetes pods being deleted/scheduled. First observation is that all pods is at Pending on those nodes, then docker ps hangs. This can only be fixed when restart the VM.

@ntquyen I've opened https://github.com/coreos/bugs/issues/1654.
